h5py/_conv.pyx

```diff
@@ -121,12 +132,72 @@ cdef herr_t init_generic(hid_t src, hid_t dst, void** priv) except -1:
     priv[0] = sizes
     sizes[0].src_size = H5Tget_size(src)
     sizes[0].dst_size = H5Tget_size(dst)
+    log_convert_registered(src, dst)
 
     return 0
 
 # =============================================================================
 # Vlen string conversion
 
+_H5PY_PY_TAG = str(H5PY_PYTHON_OPAQUE_TAG)  # cache this
+cdef bint _is_pyobject_opaque(hid_t obj):
+    # This complexity is needed to sure:
+    #   1) That ctag is freed
+    #   2) We don't segfault (for some reason a try-finally statement is needed,
+    #   even if we do (what I think are) the right steps in copying and freeing.
+    cdef char* ctag = NULL
+    try:
+        if H5Tget_class(obj) == H5T_OPAQUE:
+            ctag = H5Tget_tag(obj)
+            if ctag != NULL:
+                tag = str(ctag)
```

Comment:
Is it practical to compare the `char*` strings in C (`strcmp`?), rather than as Python objects?

Reply:
That may work better, I'll try that.